packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1318,47 +1349,138 @@ export class EthModule {
     this._logger.log(util.inspect(error));
   }
 
-  private _logFrom(from: Buffer) {
-    this._logger.logWithTitle("From", bufferToHex(from));
+  private _logFrom(from: Buffer, indent = false) {
+    this._logger.logWithTitle("From", bufferToHex(from), indent);
   }
 
   private async _sendTransactionAndReturnHash(tx: Transaction) {
-    const result = await this._node.sendTransaction(tx);
+    const results = await this._node.sendTransaction(tx);
     const txHash = bufferToRpcData(tx.hash(true));
-    if (result === undefined) {
+    if (results === undefined) {
       return txHash;
     }
 
-    const { trace, block, blockResult, consoleLogMessages, error } = result;
-
-    await this._logTransactionTrace(tx, trace, block, blockResult);
+    let multipleTransactions = false;
+    let sentTxResult: RunTransactionResult | undefined;
+    let sentTxIndex: number | undefined;
+    let error: Error | undefined;
 
-    if (trace !== undefined) {
-      await this._runHardhatNetworkMessageTraceHooks(trace, false);
+    if (results.length > 1) {
+      this._logMultipleBlocksWarning();
     }
 
-    this._logConsoleLogMessages(consoleLogMessages);
+    for (const result of results) {
+      const { block, blockResult, traces } = result;
+      let additionalIndent = false;
+
+      if (results.length === 1 && block.transactions.length > 1) {
+        this._logMultipleTransactionsWarning();
+        this._logBlockNumber(block);
+        multipleTransactions = true;
+        additionalIndent = true;
+      } else if (results.length > 1) {
+        this._logBlockNumber(block);
+        multipleTransactions = true;
+        additionalIndent = true;
+      }
 
-    if (error !== undefined) {
-      if (this._throwOnTransactionFailures) {
-        throw error;
+      for (let i = 0; i < block.transactions.length; i++) {
+        const blockTx = block.transactions[i];
+        const trace = traces[i];
+        let isSentTx = false;
+
+        if (bufferToRpcData(blockTx.hash(true)) === txHash) {
```

Comment:
`true` is the default argument, so no need to write it
```suggestion
        if (bufferToRpcData(blockTx.hash()) === txHash) {
```